packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js

```diff
@@ -105,31 +111,42 @@ class CryptoWatchPriceFeed extends PriceFeedInterface {
 
     // If the time is before the first piece of data in the set, return null because
     // the price is before the lookback window.
-    if (time < firstPricePeriod.openTime) {
+    if (time < firstPricePeriod.openTime - this.historicalTimestampBuffer) {
       throw new Error(`${this.uuid}: time ${time} is before firstPricePeriod.openTime`);
     }
 
     // historicalPricePeriods are ordered from oldest to newest.
     // This finds the first pricePeriod whose closeTime is after the provided time.
     const match = this.historicalPricePeriods.find((pricePeriod) => {
-      return time < pricePeriod.closeTime && time >= pricePeriod.openTime;
+      return (
+        time < pricePeriod.closeTime + this.historicalTimestampBuffer &&
+        time >= pricePeriod.openTime - this.historicalTimestampBuffer
+      );
     });
 
+    // TODO: Add comments why `match` could be undefined, but the following code within the `if` statement
```

Comment:
@UMAprotocol/eng any idea why the below `match === undefined` situation could occur? The matching logic looks exactly the same as in above except for the `reverse()` call.

Reply:
`match === undefined` happens when no candle is found where the timestamp falls in the range above.

Inside the if statement, there are two separate finds.

The first searches the array in reverse to find the last candle (or first in the reversed array) that falls before the timestamp in question. This is only half of the condition in the `find` above.

The second searches the array forwards to find the first candle that falls after the timestamp in question. Once again, this is only half of the inequality above.

I think the code formatting can be a little confusing, as it might make it look like these are one `find` call rather than two separate ones.

Reply:
Ah I see all the logic inside the `if` statement is just aimed to provide the caller with more details